lib/plugins/aws/provider/awsProvider.js

```diff
@@ -114,6 +114,16 @@ const impl = {
       impl.addProfileCredentials(results, profile);
     }
   },
+  /**
+   * Check the metadata endpoint to see if we've reverted to IAM foles.
+   */
+  checkMetadataEndpoint: () => {
+    const sts = new AWS.STS()
+    sts.getCallerIdentity((err) => {
```

Comment:
Could use async-await here to use getCallerIdentity(), even though a synchronous return is needed for the call to checkMetadataEndpoint?